### PR TITLE
Add Surefire Max Heap Size argument

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Remove Travis Maven overrides. Fork in Travis CI.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @dhalperi

File: runners/flink/pom.xml [hunk 1]
```diff
@@ -82,7 +82,7 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-surefire-plugin</artifactId>
           <configuration>
-            <argLine>-Dlog4j.configuration=log4j-test.properties  -XX:-UseGCOverheadLimit</argLine>
+            <argLine>-Dlog4j.configuration=log4j-test.properties  -XX:-UseGCOverheadLimit -Xmx512m</argLine>
```
Comment (dhalperi) [inline runners/flink/pom.xml]: The hope:

* Define a `-Dbeam.argline=....` ENV in `.travis.yml`
* Dereference `${beam.argline}` in the three places it needs to be right now.
* Separate out the possibility of cleaning up flink override of argline.

Comment (tgroh) [inline runners/flink/pom.xml]: Done.

**Comment by dhalperi** [comment]
LGTM

**Comment by dhalperi** [comment]
Merged.

Please backport in spirit to Dataflow.

**Pull request closed without merging**